addon/-private/themes/default-theme.js

```diff
@@ -0,0 +1,29 @@
+export default {
+  table: 'yeti-table',
+
+  row: '',
+
+  thead: '',
+  theadRow: '',
+  theadCell: '',
+
+  tbodyRow: '',
+  tbodyCell: '',
+
+  tfoot: '',
+  tfootRow: '',
+  tfootCell: '',
+
+  columnSortable:  'yeti-table-sortable',
+  columnSorted: 'yeti-table-sorted',
+  columnSortedAsc: 'yeti-table-sorted-asc',
+  columnSortedDesc: 'yeti-table-sorted-desc',
+
+  pagination: {
```

Comment:
My last concern is the `pagination` object.
With the current way this is set up, if we do something like:

```hbs
<YetiTable @theme={{hash pagination=(hash info="some-class)}}>
```

all the other properties (pageSize, next and previous) would be overwritten be this new object.

One solution would be to always have a flat theme object structure, probably namespaced.
```js
paginationInfo: 'yeti-table-pagination-controls-page-info',
paginationSize: 'yeti-table-pagination-controls-page-size',
paginationNext: 'yeti-table-pagination-controls-next',
paginationPrevious: 'yeti-table-pagination-controls-previous'
```

But I do see how having some structure could help tidying up things a bit. In that case we would have to look for a way to "Object.assign" without overwriting the entire object. There are some packages on npm that do that.

What are your thoughts on this?

Reply:
I think the original reason I was attempting Theme as an actual emberObject was to use mergedProperties in ember to solve this.

I prefer the organization of the nested objects and could see users arbitrarily adding nested objects to the config version and then trying to merge them, so it would be nice if it supported this.   

Was searching and thought this one looked like it would do the trick https://davidwalsh.name/javascript-deep-merge (https://github.com/TehShrike/deepmerge)

Reply:
That one looks great! `deepMerge.all()` seems to be what we're looking for.

With ember-auto-import in dependências we should be able to import it.

Reply:
Ill see what I can do later. I thought about putting the 4 sort ones under a nest as well.

Reply:
That sounds good.